scripts/fit.py

```diff
@@ -236,29 +236,36 @@ def jac(x, pars):
         options = dict(maxiter=1e4, gtol=1e-3)
 
     def _fit(N_PARS=3, sign=1):
+        bounds = None
+        if method in ["Nelder-Mead"]:
+            bounds = ((-1e9, 1e9), (-1e9, 0), (0, 10))
```

Comment:
Why is it only for Nelder-Mead? I understand that setting bounds for BFGS and trust-exact don't do anything, but does it actually crash when you set bounds? If not, I think the restriction to Nelder-Mead only can be relaxed.

Reply:
Also, I would maybe allow also negative values for the c-Parameter. I mean we have seen in the past that due to bad MC stitching or bugs in the event generation we got some weird offsets. In those cases the fit would probably just had crashed with those bounds and there would be no debug plots. So I would also relax the bounds to cover at least also some unexpected cases.

Reply:
How do you expect negative c-values? That would mean that we have a negative cross-section. This should never be the case, except when maybe looking into a single negatively interfering loop?

Reply:
> Why is it only for Nelder-Mead? I understand that setting bounds for BFGS and trust-exact don't do anything, but does it actually crash when you set bounds? If not, I think the restriction to Nelder-Mead only can be relaxed.

Ok, I did not know this. I will change it.

Reply:
It could happen that for some reason there are a lot of events with negative event weights in certain parts of phase space. Of course, this shouldn't happen because that would mean there is something wrong with the MC event generation, but in order to spot such bugs, I think one could relax these boundaries a bit. Or maybe even better, change the plotting script that it plots the individual bins nonetheless even when the fit fails, but doesn't write the fit results json so the law task is not marked as successful. This is perhaps the better solution, but would require a bit more work in the PlotNPCorr task.

Reply:
Fixed